proarc-webapp/src/main/java/cz/cas/lib/proarc/webapp/server/rest/WorkflowResource.java

```diff
@@ -483,6 +495,15 @@ public WorkflowResource(
         for (JobDefinition job : workflowDefinition.getJobs()) {
             if ((name == null || name.equals(job.getName()))
                     && (disabled == null || disabled == job.isDisabled())) {
+                Set<String> collection = MetaModelRepository.getInstance().find()
+                        .stream().map(metaModel -> metaModel.getPid()).collect(Collectors.toSet());
+
+                for (ModelDefinition model : job.getModel()) {
+                    if (!collection.contains(model.getType())) {
+                        return SmartGwtResponse.asError(WorkflowProfileConsts.MODEL_TYPE + " - invalid value! " + model.getType());
+                    }
+                }
```

Comment:
Ne tak docela :-) Potřeboval jsem vypsat špatné modely, resp. tato implementace stejně vypsala jen jeden....Přepsal jsem to do stream api, tak to bude lepší.